### PR TITLE
Blacklist invalid IPs in faraday

### PR DESCRIPTION
**Pull request by samn**

### CODE REVIEW

File: lib/restrict_ip_addresses.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+require 'faraday/middleware'
+require 'ipaddr'
+require 'uri'
+
+BLACKLISTED_IP_RANGES = [
+  '169.254.169.254',
```
Comment (kb) [inline lib/restrict_ip_addresses.rb]: /32 for completeness?

**Comment by kb** [comment]
Doesn't feel great that we are resolving a real domain in unit tests. Is there another way?

**Comment by samn** [comment]
Stubbed out the resolv

File: lib/restrict_ip_addresses.rb (hunk 1 above)
Comment (samn) [inline lib/restrict_ip_addresses.rb]: fixed

File: lib/service/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Service
-  VERSION = '3.30.0'
+  VERSION = '3.31.0'
```
Comment (jladieu) [inline lib/service/version.rb]: We've got a version collision between our two PRs, so let's either combine them or sequence them.

File: lib/restrict_ip_addresses.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      super(app)
+    end
+
+    def call(env)
+      raise AddressNotAllowed.new('Invalid Address') if denied?(env)
+      @app.call(env)
+    end
+
+    def denied?(env)
+      addresses(env[:url].host).any? { |a| denied_ip?(a) }
```
Comment (jladieu) [inline lib/restrict_ip_addresses.rb]: What's the class that you're expecting to inhabit `env[:url]`? -- will it definitely have a `host` method you can call?  How about the `nil` case?

Comment (samn) [inline lib/restrict_ip_addresses.rb]: Faraday populates this, i think it's always present
https://github.com/lostisland/faraday#writing-middleware

**Comment by jladieu** [comment]
Looks great, thanks for doing this!

**Pull request merged**